src/controllers/category-controller.js

```diff
@@ -1,4 +1,29 @@
 // eslint-disable-next-line no-unused-vars
 const { CategoryModel } = require('../models');
+const { NotFoundError } = require('../errors');
 
-module.exports = {};
+async function getCategories(req, res, next) {
+  try {
+    const { limit, skip } = req.query;
+
+    const [ categories, total ] = await Promise.all([
+      CategoryModel.find().limit(parseInt(limit)).skip(parseInt(skip)),
+      CategoryModel.countDocuments()
+    ]);
+
+    if (!categories) {
```

Comment:
in lists we don't return not found error, we just send emtpy array that was returned from db